### DISCUSSION TITLE
Allow to Sort by Select Column

### DISCUSSION THREAD

**Discussion by RHeynsZa** [KevinVandy/material-react-table#692] [Ideas]
This would be a super cool feature to add. We have added it manually, but this implemetation is a bit buggy when it comes to filtering on the rows.

![image](https://github.com/KevinVandy/material-react-table/assets/58415372/1c98f900-0d78-48fb-a733-3c9f7f706e2f)

Sort by selection doesnt seem too intense, so I could probably contribute this if its deemed okay?

**Comment by Kiliusz** [comment]
I had implement it in standalone react-table. @RHeynsZa  Can you provide your solution?

**Comment by RHeynsZa** [comment]
Ugh, we ripped it out a while ago. We wanted some other functionality, and maintaining the custom solution was a bit of a headache for the value that it added

As for implementation wise,
But, we added our own custom select column, with our own checkbox. 

Then we keep our own selections state. Once the selection updates, we sort our custom state, and update the table. Not clean at all, but it worked pretty well

**Comment by Skippidar** [comment]
I've found this information in documentation: https://www.material-react-table.com/docs/guides/display-columns#display-column-definition-options-prop

![CleanShot 2024-03-21 at 12 27 04@2x](https://github.com/KevinVandy/material-react-table/assets/322145/8aeb60d8-8d5e-42c0-8d17-3519cd9cdd19)

There is information about enabling actions, ordering, etc. BUT such options are missing in MRT_DisplayColumnDef type and they don't work. I can't find out if it's planned or maybe just wrong information in documentation.

P.S.: Sorry for duplicating message, but there are 2 discussions about same feature from you
